pkg/webhook/resources/setting/validator.go

```diff
@@ -213,19 +223,81 @@ func validateDeleteSetting(oldObj runtime.Object) error {
 	return nil
 }
 
-func validateHTTPProxy(setting *v1beta1.Setting) error {
+func (v *settingValidator) validateHTTPProxy(setting *v1beta1.Setting) error {
 	if setting.Value == "" {
 		return nil
 	}
-	if err := json.Unmarshal([]byte(setting.Value), &util.HTTPProxyConfig{}); err != nil {
+
+	httpProxyConfig := &util.HTTPProxyConfig{}
+	if err := json.Unmarshal([]byte(setting.Value), httpProxyConfig); err != nil {
 		message := fmt.Sprintf("failed to unmarshal the setting value, %v", err)
 		return werror.NewInvalidError(message, "value")
 	}
+
+	// Make sure the node's IP addresses is set in 'noProxy'. These IP
+	// addresses can be specified individually or via CIDR address.
+	nodes, err := v.nodeCache.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	err = validateNoProxy(httpProxyConfig.NoProxy, nodes)
+	if err != nil {
+		return err
+	}
```

Comment:
Should they use `werror.NewInvalidError` to create validation error as well?

Reply:
> Should they use `werror.NewInvalidError` to create validation error as well?

The function `validateNoProxy` returns `NewInvalidError` errors. But you are right for the `nodes, err := v.nodeCache.List(labels.Everything())` error case.